main.nf

```diff
@@ -46,39 +72,128 @@ def helpMessage() {
-if (params.fasta) { ch_fasta = file(params.fasta, checkIfExists: true) }
+params.gtf = params.genome ? params.genomes[ params.genome ].gtf ?: false : false
+params.star_index = params.genome ? params.genomes[ params.genome ].star ?: false : false
+
+// Check input path parameters to see if they exist
+checkPathParamList = [
+    params.input,
+    params.fasta,
+    params.gtf,
+    params.star_index,
+    params.smrna_fasta,
+    params.fai
+]
+for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
+
+if(!params.smrna_fasta) {
+    if(params.genome) {
+        log.warn "There is no available smRNA fasta file associated with the provided genome '${params.genome}'; pre-mapping will be skipped. A smRNA fasta file can be specified on the command line with --smrna_fasta"
+    } else {
+        log.warn "There is no smRNA fasta file suppled for genome specified; pre-mapping will be skipped. A smRNA fasta file can be specified on the command line with --smrna_fasta or --smrna_org"
+    }
+}
+
+// Set up peak caller logic
+def paraclu_check = false
+def icount_check = false
+def pureclip_check = false
+def piranha_check = false
+
+if (params.peakcaller){
+
+    def peak_list = params.peakcaller.split(',').collect()
+    peak_list.each {
+        if ( it == 'all') {
+            paraclu_check = true
+            icount_check = true
+            pureclip_check = true
+            piranha_check = true
+        } else if ( it == 'paraclu' && !paraclu_check ) {
+            paraclu_check = true
+        } else if ( it == 'icount' && !icount_check ) {
+            icount_check = true
+        } else if ( it == 'pureclip' && !pureclip_check ) {
+            pureclip_check = true
+        } else if ( it == 'piranha' && !piranha_check ) {
+            piranha_check = true
+        } else {
+            exit 1, "Invalid peak caller option: ${it}. Valid options: 'icount', 'paraclu', 'pureclip', 'piranha'"
+        }
+    }
+}
+
+// cannot run icount wihtout gtf file
+if (!params.gtf && icount_check) {
+    icount_check = false
+    log.warn "iCount can only be run with a gtf annotation file - iCount will be skipped"
+}
+
+def gtf_check = false
+String gtf_file_str = ""
+String gtf_col_3 = ""
+if (params.gtf && icount_check) {
+    if (hasExtension(params.gtf, 'gz')) {
+        gtf_file_str = "${workflow.workDir}/tmp_gtf.txt"
+        decompressGzipFile(params.gtf, gtf_file_str)
+    } else {
+        gtf_file_str = params.gtf
+    }
+    File gtf_file = new File(gtf_file_str)
+    boolean compatibility = check_gtf_by_line( gtf_file, 30 )
+    if (hasExtension(params.gtf, 'gz')) {
+        boolean fileSuccessfullyDeleted =  new File("${workflow.workDir}/tmp_gtf.txt").delete()
+    }
+    if (compatibility) {
+        gtf_check = true
+    }
+    if (!gtf_check) {
+        icount_check = false
+        log.warn "The supplied gtf file is not compatible with iCount. Peakcalling with iCount will be skipped"
+    }
+}
```

Comment:
In this bit we're reading the gtf to check if it meets criteria that make it compatible with the icount peakcaller, so that if it's not compatible then we wont run the icount process. Therefore we'd need some kind of way for the wrapping process to communicate this to an outside parameter so it can control whether icount runs - is this possible?

Reply:
Okay this has been completely removed now